src/routes/safes/safes.service.ts

```diff
@@ -141,33 +137,48 @@ export class SafesService {
     return lastQueuedTransaction.results[0]?.modified ?? null;
   }
 
-  private async executedTransactionTag(
+  /**
+   * Gets the txHistoryTag date, i.e. the modification/submission/execution date for
+   * the most recent transaction (multisig, module, transfer) associated to the {@link safeAddress}
+   *
+   * For multisig transactions, 'modified' (or 'submissionDate' if absent) is taken.
+   * For module transactions and transfers, 'executionDate' is taken.
+   *
+   * @param chainId - chain id to use
+   * @param safeAddress - the address of the target Safe
+   * @returns {@link Date} the modification/submission/execution date for the most recent transaction
+   */
+  private async getTxHistoryTagDate(
     chainId: string,
     safeAddress: string,
   ): Promise<Date | null> {
-    const lastExecutedTransaction = (
-      await this.safeRepository.getTransactionHistoryByExecutionDate({
+    const txPages = await Promise.all([
+      this.safeRepository.getMultisigTransactions({
         chainId,
         safeAddress,
         limit: 1,
-      })
-    ).results[0];
-
-    if (!lastExecutedTransaction) return null;
-
-    if (isMultisigTransaction(lastExecutedTransaction)) {
-      return (
-        lastExecutedTransaction.modified ??
-        lastExecutedTransaction.submissionDate
-      );
-    } else if (isEthereumTransaction(lastExecutedTransaction)) {
-      return lastExecutedTransaction.executionDate;
-    } else if (isModuleTransaction(lastExecutedTransaction)) {
-      return lastExecutedTransaction.executionDate;
-    } else {
-      // This should never happen as AJV would not allow an unknown transaction to get to this stage
-      throw Error('Unrecognized transaction type');
-    }
+        executed: true,
+      }),
+      this.safeRepository.getModuleTransactions({
+        chainId,
+        safeAddress,
+        limit: 1,
+      }),
+      this.safeRepository.getTransfers({
+        chainId,
+        safeAddress,
+        limit: 1,
+      }),
+    ]);
+
+    const dates = txPages
+      .flatMap(({ results }) => [...results])
```

Comment:
I had to fiddle with the types a bit, but this satisfies TypeScript, as well as removes the unnecessary spread:

```suggestion
      .flatMap(({ results }): (MultisigTransaction | ModuleTransaction | Transfer)[] =>
        results
      )
```